File: smart-scheduler/backend/Models/Contractor.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartScheduler.Models;

public class Contractor
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public ContractorType Type { get; set; }
    public decimal Rating { get; set; } = 0.00m;
    public ContractorStatus Status { get; set; } = ContractorStatus.Active;
    public string PhoneNumber { get; set; } = string.Empty;
    public string Email { get; set; } = string.Empty;
    
    // Value object - owned entity
    public BaseLocation BaseLocation { get; set; } = new();
    
    // Skills stored as JSON array
    public List<string> Skills { get; set; } = new();
    
    // Working hours - separate table
    public List<ContractorWorkingHours> WorkingHours { get; set; } = new();
    
    // Audit columns
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    
    // Optimistic concurrency control
    [Timestamp]
    public byte[] RowVersion { get; set; } = Array.Empty<byte>();
}

File: smart-scheduler/frontend/services/types/contractor.ts
// Contractor types matching backend DTOs
export enum ContractorType {
  Flooring = 1,
  Tile = 2,
  Carpet = 3,
  Multi = 4,
}

export enum ContractorStatus {
  Active = 1,
  Inactive = 2,
  OnLeave = 3,
}

export interface BaseLocation {
  latitude: number;
  longitude: number;
  address: string;
  city: string;
  state: string;
  zipCode: string;
}

export interface WorkingHours {
  dayOfWeek: number; // 0 = Sunday, 1 = Monday, ..., 6 = Saturday
  startTime: string; // "HH:mm" format
  endTime: string; // "HH:mm" format
}

export interface Contractor {
  id: string;
  name: string;
  type: ContractorType;
  rating: number;
  status: ContractorStatus;
  phoneNumber: string;
  email: string;
  baseLocation: BaseLocation;
  skills: string[];
  workingHours: WorkingHours[];
  availabilityStatus?: string;
  statistics?: ContractorStatistics;
  createdAt: string;
  updatedAt: string;
  rowVersion?: string; // Base64 encoded for concurrency control
}

export interface ContractorStatistics {
  totalJobs: number;
  completedJobs: number;
  pendingJobs: number;
  averageRating: number;
}

export interface ContractorListItem {
  id: string;
  name: string;
  type: ContractorType;
  rating: number;
  status: ContractorStatus;
  phoneNumber: string;
  email: string;
  city: string;
  state: string;
  availabilityStatus: string;
}

export interface CreateContractorRequest {
  name: string;
  type: ContractorType;
  phoneNumber: string;
  email: string;
  baseLocation: BaseLocation;
  skills?: string[];
  workingHours?: WorkingHours[];
}

export interface UpdateContractorRequest {
  name?: string;
  type?: ContractorType;
  phoneNumber?: string;
  email?: string;
  baseLocation?: BaseLocation;
  skills?: string[];
  workingHours?: WorkingHours[];
  status?: ContractorStatus;
  rowVersion?: string;
}

export interface PagedResult<T> {
  data: T[];
  totalCount: number;
  currentPage: number;
  pageSize: number;
  totalPages: number;
}

export interface ListContractorsParams {
  name?: string;
  type?: ContractorType;
  minRating?: number;
  maxRating?: number;
  city?: string;
  state?: string;
  includeInactive?: boolean;
  page?: number;
  pageSize?: number;
}

